jupyter_core/command.py

```diff
@@ -113,9 +115,24 @@ def _execvp(cmd, argv):
         os.execvp(cmd, argv)
 
 
+def _jupyter_realpath(path, subcommand):
+    # get currrent PATH
+    _path_env = os.environ['PATH']
+    # set new PATH with self
+    os.environ['PATH'] = os.pathsep.join(_path_with_self())
+    # get the real path for the jupyter-<subcommand>
+    real_path = find_executable('{}-{}'.format(path, subcommand))
+    # RESTORE previous PATH
+    os.environ['PATH'] = _path_env
+    return (
+        real_path if real_path is not None and os.access(real_path, os.X_OK)
+        else None
```

Comment:
We'll probably get more meaningful error messages if we throw an error when we can't find the command rather than returning None.